### ISSUE TITLE
Measure distance between meshes

### ISSUE THREAD

**Issue by PedroLopes211** [mikedh/trimesh#777]
For my project, I need to measure the distance between two STL files. I wrote a script that allows reading the files, positioning them in relation to each other in the desired position. Now, in the next step I need to check the distance from one object to the other. Is there a function or script available in this library that allows me to carry out this process? Because then I’m going to want to define metrics like interpenetration area, maximum negative distance etc etc so I need to check first the distance between those objects and see if there is like mesh intersection and mesure that distance. I put the url for the combination of the 2 objects that I want to mesure the distance:

https://imgur.com/wgNaalh

**Comment by mikedh** [comment]
Hey, you could try `mesh.nearest.signed_distance`:

```
In [18]: a = trimesh.primitives.Sphere(radius=1.0, center=[0,0,0])

In [19]: b = trimesh.primitives.Sphere(radius=1.0, center=[2,0,0])

In [20]: a.nearest.signed_distance?
Signature: a.nearest.signed_distance(points)
Docstring:
Find the signed distance from a mesh to a list of points.

* Points OUTSIDE the mesh will have NEGATIVE distance
* Points within tol.merge of the surface will have POSITIVE distance
* Points INSIDE the mesh will have POSITIVE distance

Parameters
-----------
points : (n,3) float, list of points in space

Returns
----------
signed_distance : (n,3) float, signed distance from point to mesh
File:      /dropbox/drop/Dropbox/robotics/trimesh/trimesh/proximity.py
Type:      method

In [21]: a.nearest.signed_distance(b.vertices.copy()).max()
Out[21]: 0.0

```

**Comment by PedroLopes211** [comment]
I tried what you said and I get and erro to install rtree (on the image). I already install conda, got there the libspatialindex and get always the same error...
![Captura de ecrã 2020-04-14, às 11 06 58](https://user-images.githubusercontent.com/63504801/79212862-2317f200-7e40-11ea-86d8-6003a3a3c7e0.png)

**Comment by mikedh** [comment]
Did you try `conda install -c conda-forge rtree`?

**Comment by PedroLopes211** [comment]
Hello,

I already fix my problem and I actually can run the program! I google for a method and found one that worked out for me. At least I run the script to test and it was OK. (the one you wrote). I tried to run with my solids (as my object 1 is the "a" and the object 2 is the "b") and he returns me the nearest distance between them. 

Is there a way to quantify the distance between each triangle of the two objects? what I mean is, let's imagine that I cross a line between each triangle of object 2 (normal of each triangle for ex), if that line is projected on the triangles of object 1 (discovering the point of intersection with the surface), quantify that distance? Because in this case, what I did was to find the closest distance from a list points with a mesh. But imagining that there are several penetrations, if it was possible to know in which coordinates these intersections occur?
